File: scope106/src/main/java/com/example/scope106/task5/Activity.java
package com.example.scope106.task5;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.scope106.R;
import com.example.scope106.databinding.Layout5Binding;

public class Activity extends AppCompatActivity implements View.OnClickListener {
    Layout5Binding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = Layout5Binding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.editTextInput.setOnClickListener(this);
        binding.button.setOnClickListener(this);
    }

    public void enqueueWork(View v) {
        String input = binding.editTextInput.getText().toString();

        Intent serviceIntent = new Intent(this, ExampleJobIntentService.class);
        serviceIntent.putExtra("inputExtra", input);

        ExampleJobIntentService.enqueueWork(this, serviceIntent);
    }

    @Override
    public void onClick(View v) {
        if(v.getId() == R.id.button){
            enqueueWork(v);
        }
    }
}

File: scope106/src/main/java/com/example/scope106/task5/Activity.kt
package com.example.scope106.task5

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.scope106.R
import com.example.scope106.databinding.Layout5Binding

class Activity : AppCompatActivity(), View.OnClickListener {
    var binding: Layout5Binding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = Layout5Binding.inflate(layoutInflater)
        setContentView(binding!!.root)
        binding!!.editTextInput.setOnClickListener(this)
        binding!!.button.setOnClickListener(this)
    }

    fun enqueueWork(v: View?) {
        val input = binding!!.editTextInput.text.toString()
        val serviceIntent = Intent(this, ExampleJobIntentService::class.java)
        serviceIntent.putExtra("inputExtra", input)
        ExampleJobIntentService.Companion.enqueueWork(this, serviceIntent)
    }

    override fun onClick(v: View) {
        if (v.id == R.id.button) {
            enqueueWork(v)
        }
    }
}